internal/discovery/types.go

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright 2023 The Kubernetes Authors All rights reserved.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package discovery
+
+import (
+	"time"
+
+	"k8s.io/client-go/discovery"
+)
+
+// GVKMaps is a struct that holds:
+// * discovery client to fetch objects
+// * Group to Version map
+// * Group to Kind map
+// * Version to Kind map
+// * Kind to Version map
+// * Heartbeat to track the last fetch
+type GVKMaps struct {
+	Client    *discovery.DiscoveryClient
```

Comment:
It would be cleaner to have Client and Heartbeat in a separate structure as the map should just be the map and anything related to discovery & update should be handled elsewhere.